src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -739,5 +742,40 @@ public static ITypeSymbol WithNullableAnnotationFrom(this ITypeSymbol type, ITyp
 
             return symbol;
         }
+
+        public static ImmutableArray<IMethodSymbol> GetLocalFunctionSymbols(
+            this ITypeSymbol? typeSymbol,
+            ISyntaxFactsService syntaxFacts,
+            Compilation compilation,
+            CancellationToken cancellationToken)
+        {
+            if (typeSymbol is null || compilation.Language != LanguageNames.CSharp)
+            {
+                return ImmutableArray<IMethodSymbol>.Empty;
+            }
+
+            var members = typeSymbol.GetMembers();
+            using var _ = ArrayBuilder<IMethodSymbol>.GetInstance(out var builder);
+
+            foreach (var member in members)
+            {
+                foreach (var syntaxReference in member.DeclaringSyntaxReferences)
+                {
+                    var semanticModel = compilation.GetSemanticModel(syntaxReference.SyntaxTree);
+                    var node = syntaxReference.GetSyntax(cancellationToken);
+
+                    foreach (var localFunction in node.DescendantNodes().Where(syntaxFacts.IsLocalFunctionStatement))
```

Comment:
i feel ike this might double count local functions found inside accessors.   because we'll hit the prop and the accessor.